src/main/java/com/toptrumps/cli/CommandLineUI.java

```diff
@@ -49,4 +55,31 @@ public void startGame() {
         System.out.println("Thanks for playing!");
     }
 
+
+    public void askHowManyPlayers() {
+        try {
+
+            System.out.println("How many players do you want to play against?");
+            System.out.println("Please select 1-4");
+
+            int players = scanner.nextInt();
+            while (players < 1 || players > 4) {
```

Comment:
We should always avoid using literals. Although this business logic is clearly defined in the task document, this piece of information may be reused in multiple places. It would be very cumbersome if we needed to change it someday.

For this reason, I suggest extracting it to a constant. Like this:

```
private static int MIN_NUMBER_OF_PLAYERS = 1;
private static int MAX_NUMBER_OF_PLAYERS = 4;
```